### DISCUSSION TITLE
Implementing Cache Update Feature in Android Client

### DISCUSSION THREAD

**Discussion by DesarrolloAntonio** [go-shiori/shiori#821] [Q&A] [answered]
Hello,

I am currently developing a feature for updating the cache from an Android client in my app. I would appreciate some guidance on how to handle the API connections for implementing this feature. Specifically, I am looking for information on:

- The relevant API endpoints that should be used for cache management.
- Any specific request parameters required for these API calls.

Thanks.
<img src="https://github.com/go-shiori/shiori/assets/7002198/fb3208fd-8931-44fb-8a91-9d6d43f46da1" width="200">

**Comment by Monirzadeh** [comment, marked as answer]
in shiori v1.6 you can see that in `http://127.0.0.1:8080/swagger/index.html#/Auth/put_api_v1_bookmarks_cache`
in general you should send a request to the 
```
/api/v1/bookmarks/cache
```
with something like this 
```
{
  "create_archive": true,
  "create_ebook": true,
  "ids": [
    0
  ],
  "keep_metadata": true,
  "skip_exist": true
}
```
ids of bookmark you can request one id bookmark or multiple bookmark in batch mode.
if `skip_exist` be true ebook will not update if ebook file exist on server and just create ebook for bookmark  that don't have ebook. (useful when you just want request download ebook in batch mode)

**Reply by DesarrolloAntonio** [reply]
Is there any way to do it with the legacy API?

**Reply by Monirzadeh** [reply]
for endpoint in legacy you can see `legacy.go` and `home.js` here (Removed part)
https://github.com/go-shiori/shiori/pull/742/files#diff-103c3f7ac1243556d4b5447bec94fc7e8bef44d3bec5c16d940932657b313ea6
```
legacyGroup.PUT("/api/cache", r.handle(r.legacyHandler.ApiUpdateCache))
legacyGroup.PUT("/api/ebook", r.handle(r.legacyHandler.ApiDownloadEbook))
```
if i remmeber correctly
```
api/cache
```
with something like this
```
{
ids: 1,
createArchive: true,
keepMetadata: true,
createEbook: true,
}
```
(createEbook always regenerate ebook)
